node/src/blockchain/blockchain_interface.rs

```diff
@@ -2072,4 +2169,168 @@ mod tests {
 
         assert_eq!(check, vec![11, 22, 33, 44, 55, 66, 77])
     }
+
+    #[test]
+    fn non_clandestine_can_fetch_latest_block_number_successfully() {
+        let prepare_params_arc: Arc<Mutex<Vec<(String, Vec<Value>)>>> =
+            Arc::new(Mutex::new(vec![]));
+        let transport = TestTransport::default()
+            .prepare_params(&prepare_params_arc)
+            .send_result(json!("0x1e37066"));
+        let subject = BlockchainInterfaceNonClandestine::new(
+            transport.clone(),
+            make_fake_event_loop_handle(),
+            TEST_DEFAULT_CHAIN,
+        );
+
+        match subject.get_block_number() {
+            Ok(latest_block_number) => assert_eq!(latest_block_number, U64::from(31_682_662u64)),
+            Err(e) => assert_eq!(
+                e,
+                BlockchainError::QueryFailed("unexpected failure".to_string())
+            ),
+        }
+
+        let mut prepare_params = prepare_params_arc.lock().unwrap();
+        let (method_name, actual_arguments) = prepare_params.remove(0);
+        let actual_arguments: Vec<String> = actual_arguments
+            .into_iter()
+            .map(|arg| serde_json::to_string(&arg).unwrap())
+            .collect();
```

Comment:
You know, for now the assertion testing an empty vector is good enough. If it changes it will definitely breaks. But again, that parameter is so-far constant. One could point out it could change with a different version of the library. 

For clarity, I didn't complain about the fact you want to assert on this parameter I just didn't see a point in converting the potential values to a different, maybe better assertable data types. That step just prolongs the test even though it makes no difference. The vector is empty at any case.